internal/reconciler/shoot/add.go

```diff
@@ -65,11 +65,11 @@ func isRelevantShoot(obj client.Object) bool {
 	if shoot.Annotations[v1beta1constants.AnnotationAuthenticationIssuer] != v1beta1constants.AnnotationAuthenticationIssuerManaged {
 		return false
 	}
-	// Specifies whether the Shoot should be registered as a trusted cluster in the Garden cluster.
-	if shoot.Annotations[AnnotationTrustedShoot] != "true" {
-		return false
+	// Specifies whether the Shoot should be registered as a trusted cluster in the Garden cluster or to be removed from the trusted ones.
+	if shoot.Annotations[AnnotationTrustedShoot] == "true" || shoot.Annotations[AnnotationTrustedShoot] == "false" {
+		return true
 	}
-	return true
+	return false
```

Comment:
Can you clarify why was this change required? Even with the previous check the shoot will be marked as relevant during update from `true` to `false` or am I missing something? Either way I can also do an update from `true` to `foo` for example, which is not covered with this check

Reply:
My assumption is that we could later enforce only valid values (e.g., via webhook validation) so that "foo" or other arbitrary strings cannot be set.
Previously we only checked for the annotation `authentication.gardener.cloud/trusted` with value of `true` in order to trigger reconciliation.

Reply:
Yes, exactly. I think the check should be as simple as possible. We can still validate with a webhook later on so we disallow unsupported values